### PR TITLE
:sparkles: Add event dates support

### PR DESCRIPTION
**Pull request by fernandolucchesi**
Updated:

image

Old version:
A switch is used to change between events that have an exact start and end time, and events that do not.

image

image

image
(preciseTime was renamed to precise)

### CODE REVIEW

File: studio/schemas/objects/eventDate.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        timeStep: 5,
+      },
+      hidden: ({ parent }: { parent: EventDate }) => !parent.precise,
+      validation: (Rule: Rule) =>
+        Rule.custom((field: string, context: ValidationContext) => {
+          const { parent } = context as { parent: EventDate }
+          if (parent.precise && field === undefined) {
+            return 'End date is required'
+          } else if (parent.precise && parent.startDate && field <= parent.startDate) {
+            return 'End date must be greater than start date'
```
Comment (wenche) [inline studio/schemas/objects/eventDate.tsx:70]: Nice thing with an additional check here 👍

File: studio/schemas/documents/event.ts [hunk 2]
```diff
@@ -71,6 +79,12 @@ export default {
       type: 'string',
       title: 'Location',
     },
+    {
+      title: 'Date information',
```
Comment (wenche) [inline studio/schemas/documents/event.ts:83]: Not 100% sure if it needs to be a fieldset? 🤔  Might as well be without

**Review by wenche — CHANGES_REQUESTED** [review]
Looks good! 👍  Nice work. 

Just try to add the workflow we talked about in the meeting :)

**Review by adneforslund — APPROVED** [review]

File: studio/schemas/objects/eventDate.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  return !(time.includes(EMPTY) && time[0] !== time[1])
+}
+
+export default {
+  title: 'Fields for event date',
+  name: 'eventDate',
+  type: 'object',
+  fields: [
+    {
+      name: 'date',
```
Comment (SvSven) [inline studio/schemas/objects/eventDate.tsx:23]: I would add a short description where we also state the date format, just so it's clear at first glance

**Review by SvSven — COMMENTED** [review]
Looks good to me. Just to double check: did we agree to not support multi-day events? Because that doesn't seem to be possible at the moment.

**Comment by fernandolucchesi** [comment]
> Looks good to me. Just to double check: did we agree to not support multi-day events? Because that doesn't seem to be possible at the moment.

Hm that is what I understood from that meeting

**Pull request merged**